src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php

```diff
@@ -304,6 +304,19 @@ public function delete(string $id): void
         ]);
     }
 
+    public function keepalive(string $id): void
+    {
+        if (null === $this->configuration['visibility_timeout']) {
+            return;
+        }
+
+        $this->client->changeMessageVisibility([
+            'QueueUrl' => $this->getQueueUrl(),
+            'ReceiptHandle' => $id,
+            'VisibilityTimeout' => $this->configuration['visibility_timeout'],
```

Comment:
I'm not sure if we should use `visibility_timeout` here or would it be better to use the alarm interval and some fixed multiplier, f.e. `$alarmInterval * 1.2` :thinking:

Or if we use `visibility_timeout`, then maybe we could warn if `visibility_timeout` is smaller than alarm interval.

What do you think?

Reply:
Let's indeed check that `visibility_timeout` is smaller than the alarm interval.

Reply:
The cleanest I can make this happen is to add `?int $seconds = null` parameter to every `keepalive` method in the chain. It makes a lot of sense to tell the transport for how long at minimum the message should be kept alive – but it means I have to touch `keepalive()` in `Worker`, `KeepaliveReceiverInterface`, `BeanstalkdReceiver` & `BeanstalkdTransport`  too. Also need to add `Application#getAlarmInterval()`.

The changes are ready, but I think it's better to do that in separate PR as it touches so many places. I'm proposing to merge this as-is after all and I'll open a new one for this keepalive ttl where we can discuss more if needed.

Reply:
Can you please open the PR as it's a prerequisite for this one?

Reply:
Sure, here: #58552

Reply:
Now that #58552 is merged, can you rebase this one? Thank you.